serverbrowser/server.go

```diff
@@ -39,29 +40,75 @@ const (
 	LimitResultCountOption  = 1 << 7 // 0x80 / 128
 )
 
-func popString(buffer []byte, index int) (string, int) {
-	str := common.GetString(buffer[index:])
-	return str, index + len(str) + 1
+func popString(buffer []byte, index int) (string, int, error) {
+	if index < 0 || index >= len(buffer) {
+		return "", 0, errors.New("index is out of bounds")
+	}
+
+	str, err := common.GetString(buffer[index:])
+
+	if err != nil {
+		return "", 0, err
+	}
+
+	return str, index + len(str) + 1, nil
 }
 
-func popBytes(buffer []byte, index int, size int) ([]byte, int) {
-	return buffer[index : index+size], index + size
+func popBytes(buffer []byte, index int, size int) ([]byte, int, error) {
+	bufferLen := len(buffer)
+
+	if index < 0 || index >= bufferLen {
+		return nil, 0, errors.New("index is out of bounds")
```

Comment:
`errors.New("index is out of bounds")` is used in multiple places, make it a variable so you aren't initializing a new error in every control path.

Reply:
Resolved in commit [a696e6e](https://github.com/WiiLink24/wfc-server/pull/22/commits/a696e6ed9e67c42384140be045fe45069eeaca3b).